### PR TITLE
build: Publish GitHub tag in release script

### PR DESCRIPTION
**Pull request by brianluong**
# Description

As discussed in launch last Friday

## Type of change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] 😎 New feature (non-breaking change which adds functionality)
- [ ] ⁉️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] ⚒️ Refactor (no functional changes)
- [ ] 📖 Documentation (updating or adding docs)

# How Has This Been Tested?

image

image

Also 

Commented parts of the script out to test IsProductionRelease

```
➜  node-gateway git:(publish_github_tag) ✗ ./scripts/release.sh b9147f3309d8cdcd09533265016c3aaa8dc78735 0.1.0 dianwen/satsuma-gateway-test
➜  node-gateway git:(publish_github_tag) ✗ ./scripts/release.sh b9147f3309d8cdcd09533265016c3aaa8dc78735 0.1.0 satsumaxyz/node-gateway

Have you already tested this image by pushing it up to the satsumaxyz/node-gateway Docker registry? [y/N] y
```

### CODE REVIEW

File: scripts/release.sh [hunk 1]
```diff
@@ -91,6 +96,14 @@ make VERSION=$RELEASE_VERSION GIT_COMMIT_HASH=$COMMIT_HASH build-archives
 LogStep "Building and pushing the image."
 make GIT_COMMIT_HASH=$COMMIT_HASH DOCKER_HUB_REPO=$DOCKER_HUB_REPO VERSION=$RELEASE_VERSION build-and-push-image
 
+if IsProductionRelease
```
Comment (brianluong) [inline scripts/release.sh:99]: Added this flag to push tag only on production release

**Comment by brianluong** [comment]
> Looks great! Mind updating the release docs?

I updated this doc here, but the update was very small: https://github.com/satsuma-xyz/node-gateway/pull/51#issuecomment-1231847939

"After the release script finishes, create a GitHub release for the commit with the version as the tag (following the format v0.1.0). **This tag was pushed in the release script.** Upload the archives created by the release script and write release notes."

**Bold** = new stuff

File: scripts/release.sh [hunk 2]
```diff
@@ -58,9 +63,9 @@ fi
 #   exit 1
 # fi
 
-if [ "$DOCKER_HUB_REPO" == "$PRODUCTION_DOCKER_HUB_REPO" ]
+if IsProductionRelease
```
Comment (dianwen) [inline scripts/release.sh:66]: 👌 ty for the fix

**Review by dianwen — APPROVED** [review]
Good stuff!

Should we change L113 to say something to the effect of "select the tag with the release version"? It still kinda sounds like we're creating the tag alongside the release.

**Pull request merged**